### PR TITLE
Feat add message notes

### PR DESCRIPTION
**Pull request by subject026**
This adds a notes resource/endpoint plus tests.

### CODE REVIEW

File: src/note/note.controller.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  Body,
+  // Patch,
+  Param,
+  // Delete,
+  UsePipes,
+  ValidationPipe,
+} from '@nestjs/common';
+import { NoteService } from './note.service';
+import { CreateNoteDto } from './dto/create-note.dto';
+// import { UpdateNoteDto } from './dto/update-note.dto';
```
Comment (DavidRecheni) [inline src/note/note.controller.ts]: No unused code

File: src/message/message.controller.ts [hunk 2]
```diff
@@ -3,15 +3,15 @@ import {
   Get,
   Post,
   Body,
-  Patch,
+  // Patch,
```
Comment (DavidRecheni) [inline src/message/message.controller.ts]: No unused code

File: src/note/entities/note.entity.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import {
+  Column,
+  Entity,
+  PrimaryGeneratedColumn,
+  CreateDateColumn,
+  ManyToOne,
+} from 'typeorm';
+import { Message } from '../../message/entities/message.entity';
+
+@Entity({ name: 'note' })
```
Comment (DavidRecheni) [inline src/note/entities/note.entity.ts]: I don't think the props are necessary

File: src/note/entities/note.entity.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  @Column()
+  @IsNotEmpty()
+  content: string;
+
+  @CreateDateColumn()
+  createdAt: Date;
+
+  @ManyToOne(() => Message, (message) => message.notes, {
+    onDelete: 'CASCADE',
```
Comment (DavidRecheni) [inline src/note/entities/note.entity.ts:24]: !

File: src/note/note.service.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      message,
+    });
+    return this.noteRepository.save(newNote);
+  }
+
+  async findAll() {
+    return this.noteRepository.find();
+  }
+
+  // findOne(id: number) {
```
Comment (DavidRecheni) [inline src/note/note.service.ts]: No unused code

File: test/note.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+import { MessageModule } from '../src/message/message.module';
+import { TypeOrmModule } from '@nestjs/typeorm';
+import { Message } from '../src/message/entities/message.entity';
+import { Repository } from 'typeorm';
+import { Note } from '../src/note/entities/note.entity';
+import { NoteModule } from '../src/note/note.module';
+
+describe('NoteController (e2e)', () => {
+  let app: INestApplication;
+  // let noteRepository: Repository<Note>;
```
Comment (DavidRecheni) [inline test/note.e2e-spec.ts]: No unused code

**Review by DavidRecheni — CHANGES_REQUESTED** [review]

**Comment by subject026** [comment]
@DavidRecheni I've removed the table names from the entity decorators and the unused code.

Left the `onDelete: 'CASCADE'` option set on the note entity so that any notes attached to a message are deleted when that message is deleted.

**Comment by DavidRecheni** [comment]
There's still some unused code, like in note.service

**Comment by subject026** [comment]
@DavidRecheni Alright that's gone now too!

**Review by DavidRecheni — APPROVED** [review]

**Pull request merged**